skimage/restoration/deconvolution.py

```diff
@@ -375,16 +376,33 @@ def richardson_lucy(image, psf, iterations=50, clip=True):
         convolve_method = fftconvolve
     else:
         convolve_method = convolve
-
     image = image.astype(np.float)
-    psf = psf.astype(np.float)
-    im_deconv = np.full(image.shape, 0.5)
+    psf = psf.astype(np.float)    
     psf_mirror = psf[::-1, ::-1]
-
-    for _ in range(iterations):
+    im_deconv = image
+    im_deconv_mid = np.zeros_like(image)
+    v = np.zeros_like(image)
+    #initial iteration
+    relative_blur = image / convolve_method(im_deconv, psf, 'same')
+    im_deconv_mid = im_deconv * convolve_method(relative_blur, psf_mirror,
+                                                    'same')
+    v = im_deconv_mid - im_deconv
+    np.clip(v, 0, v)           
```

Comment:
- I don't understand this line, didi you mean `np.maximum(v, 0)`?
- add `out=v` to `np.clip` call, otherwise no change occurs to `v`